core/api/src/servers/middlewares/scope.ts

```diff
@@ -3,79 +3,77 @@ import { GraphQLResolveInfo, GraphQLFieldResolver } from "graphql"
 import ScopesOauth2 from "@/domain/authorization"
 import { AuthorizationError } from "@/domain/errors"
 import { mapError } from "@/graphql/error-map"
-import { mutationFields } from "@/graphql/public"
+import { mutationFields, queryFields } from "@/graphql/public"
 
-const readTransactionsAuthorize = async (
+const readAuthorize = async (
   resolve: GraphQLFieldResolver<unknown, GraphQLPublicContextAuth>,
   parent: unknown,
   args: unknown,
   context: GraphQLPublicContextAuth,
   info: GraphQLResolveInfo,
 ) => {
   const scope = context.scope
-  const appId = context.appId
-
-  // not a delegated token
-  if (appId === undefined || appId === "") {
-    return resolve(parent, args, context, info)
-  }
 
+  console.log("SCOPE ", scope)
+  // not a token with scope
   if (scope === undefined || scope.length === 0) {
-    return mapError(
-      new AuthorizationError("appId is defined but scope is undefined or empty"),
-    )
+    return resolve(parent, args, context, info)
   }
 
-  if (scope.find((s) => s === ScopesOauth2.TransactionsRead) !== undefined) {
+  if (scope.find((s) => s === ScopesOauth2.Read) !== undefined) {
     return resolve(parent, args, context, info)
   }
 
-  return mapError(new AuthorizationError("not authorized to read transactions"))
+  return mapError(new AuthorizationError("not authorized to read data"))
 }
 
-const paymentSendAuthorize = async (
+const writeAuthorize = async (
   resolve: GraphQLFieldResolver<unknown, GraphQLPublicContextAuth>,
   parent: unknown,
   args: unknown,
   context: GraphQLPublicContextAuth,
   info: GraphQLResolveInfo,
 ) => {
   const scope = context.scope
-  const appId = context.appId
 
-  // not a delegated token
-  if (appId === undefined || appId === "") {
+  // not a token with scope
+  if (scope === undefined || scope.length === 0) {
     return resolve(parent, args, context, info)
   }
 
-  if (scope === undefined) {
-    return mapError(new AuthorizationError("appId is defined but scope is undefined"))
-  }
-
-  if (scope.find((s) => s === ScopesOauth2.PaymentsSend) !== undefined) {
+  if (scope.find((s) => s === ScopesOauth2.Write) !== undefined) {
```

Comment:
will we get an otel span propagate with the scope of the request?

Reply:
They should be added here but I can't seem to find the information in the query https://github.com/GaloyMoney/galoy/blob/b87bfcd1539d35045551b1457ba478f8841041e2/core/api/src/servers/graphql-main-server.ts#L82